viral/single_session.py

```
@@ -386,6 +383,16 @@ def plot_speed_reward_unrewarded(
             step_size=step_size,
             sampling_rate=sampling_rate,
         )
+        for bin_idx in range(len(speed)):
+            if np.isinf(speed[bin_idx].speed):
+                if speed[bin_idx].position_start >= 180:
+                    continue  # Doesn't matter
+                print(
+                    f"Got inf speed for position {speed[bin_idx].position_stop} in trial {idx}"
+                )
+                speed[bin_idx].speed = np.mean(
+                    [bin.speed for bin in speed if not np.isinf(bin.speed)]
```

Ruff (0.16.0)

[error] Variable `bin` is shadowing a Python builtin

(A001)